test-util/data/foundation-models/azure-openai-chat-completion-stream-chunk-response-token-usage.json

```diff
@@ -0,0 +1 @@
+{"choices":[],"created":1730125149,"id":"chatcmpl-ANKsHIdjvozwuOGpGI6rygvwSJH0I","model":"gpt-35-turbo","object":"chat.completion.chunk","system_fingerprint":"fp_808245b034","usage":{"completion_tokens":7,"prompt_tokens":14,"total_tokens":21}}
```

Comment:
We can also get an error during streaming. See [here for an example](https://github.com/SAP/ai-sdk-java/blob/main/foundation-models/openai/src/test/resources/streamChatCompletionError.txt).

We should probably add a test for this case, ensuring we throw in such cases..

Reply:
I now throw an error if sse does not start with `data` or `event`. In the other case like the above link, the sse decoder treats both json as one string / one payload, and we will throw anyways while doing JSON parse in `SseStream.fromSSEResponse()`.

Reply:
I now throw an error if a new sse does not start with `data` or `event`. In case of two lines been put together like the link above, the sse decoder treats them as one single payload as there is no empty line separating them. In this case, we will throw while doing JSON parse in sse-stream.ts. Test case is added in sse-stream.test.ts.